src/NHibernate/Impl/FutureBatch.cs

```diff
@@ -73,21 +73,36 @@ private IList GetResults()
 			if (results != null)
 				return results;
 
-			var multiApproach = CreateMultiApproach(isCacheable, cacheRegion);
-			var needTransformer = false;
-			foreach (var query in queries)
+			if (!session.Factory.ConnectionProvider.Driver.SupportsMultipleQueries)
 			{
-				AddTo(multiApproach, query.Query, query.ResultType);
-				if (query.Future?.ExecuteOnEval != null)
-					needTransformer = true;
+				results = new List<object>();
+				foreach (var query in queries)
+				{
+					var result = List(query.Query);
```

Comment:
Here is another "discrepancy" that I have not listed in the commit message: without multi-queries support, it was executing the generic typed version. Due to current `FutureBatch` implementation, we are back to executing the non-generic one. This is nearer to what is done with multi-queries support, although I would prefer having both using generic methods. But there are much more changes to do in `FutureBatch` and multi queries if we want to replace non-generic usages, so I leave that for a later PR.